EventFilter/RPCRawToDigi/plugins/RPCDigiMerger.cc

```diff
@@ -39,53 +61,108 @@ void RPCDigiMerger::beginRun(edm::Run const& run, edm::EventSetup const& setup)
 
 void RPCDigiMerger::produce(edm::Event& event, edm::EventSetup const& setup) {
   // Get the digis
-  // TwinMux
-  Handle<RPCDigiCollection> TwinMux_digis;
-  event.getByToken(twinMux_token_, TwinMux_digis);
-  // OMTF
-  Handle<RPCDigiCollection> OMTF_digis;
-  event.getByToken(omtf_token_, OMTF_digis);
-  // CPFF
-  Handle<RPCDigiCollection> CPPF_digis;
-  event.getByToken(cppf_token_, CPPF_digis);
-
   // new RPCDigiCollection
   std::unique_ptr<RPCDigiCollection> rpc_digi_collection(new RPCDigiCollection());
 
-  // loop over TwinMux digis
-  for (const auto& rpcdgIt : (*TwinMux_digis)) {
-    // The layerId
-    const RPCDetId& rpcId = rpcdgIt.first;
-    // Get the iterators over the digis associated with this LayerId
-    const RPCDigiCollection::Range& range = rpcdgIt.second;
-
-    rpc_digi_collection->put(range, rpcId);
-  }
-
-  // loop over CPPF digis
-  for (const auto&& rpcdgIt : (*CPPF_digis)) {
-    // The layerId
-    const RPCDetId& rpcId = rpcdgIt.first;
-    // Get the iterators over the digis associated with this LayerId
-    const RPCDigiCollection::Range& range = rpcdgIt.second;
-
-    rpc_digi_collection->put(range, rpcId);
-  }
-
-  // loop over OMTF digis
-  for (const auto& rpcdgIt : (*OMTF_digis)) {
-    // The layerId
-    const RPCDetId& rpcId = rpcdgIt.first;
-    // Get the iterators over the digis associated with this LayerId
-    const RPCDigiCollection::Range& range = rpcdgIt.second;
-
-    // accepts only rings: RE-2_R3 ; RE-1_R3 ; RE+1_R3 ; RE+2_R3 ;
-    if (((rpcId.region() == -1 || rpcId.region() == 1) && (rpcId.ring() == 3) &&
-         (rpcId.station() == 1 || rpcId.station() == 2))) {
-      rpc_digi_collection->put(range, rpcId);
+  //Check if its Data
+  if (not(cppf_token_.isUninitialized() && omtf_token_.isUninitialized() && twinMux_token_.isUninitialized())) {
+    // TwinMux
+    Handle<RPCDigiCollection> TwinMux_digis;
+    event.getByToken(twinMux_token_, TwinMux_digis);
+    // OMTF
+    Handle<RPCDigiCollection> OMTF_digis;
```

Comment:
Move inside the check for initialization below

Reply:
ok